core/pipeline/label.go

```diff
@@ -133,29 +146,189 @@ func LoadGitLabels(workdir string) ([]Label, error) {
 	return labels, nil
 }
 
+func LoadCircleCILabels() ([]Label, error) {
+	if os.Getenv("CIRCLECI") != "true" { // nolint:goconst
+		return []Label{}, nil
+	}
+
+	labels := []Label{
+		{
+			Name:  "dagger.io/vcs.change.branch",
+			Value: os.Getenv("CIRCLE_BRANCH"),
+		},
+		{
+			Name:  "dagger.io/vcs.change.head_sha",
+			Value: os.Getenv("CIRCLE_SHA1"),
+		},
+		{
+			Name:  "dagger.io/vcs.job.name",
+			Value: os.Getenv("CIRCLE_JOB"),
+		},
+	}
+
+	appendLabel := func(label string, envVar []string, labels []Label) []Label {
```

Comment:
this function is a bit confusing to follow and to use. Maybe an idea on how we could use it and name it could be:

labels = append(labels, firstEnvLabel("dagger.io/vsc.change.number",pipelineNumber))

^ that follows the standard go convention and provides a more clear name for the function.

Reply:
I've implemented your solution in a [new commit](https://github.com/dagger/dagger/pull/5705/commits/d467fbfaa3ec4b0a13599ce59b994b38408614d2). However, to avoid adding empty `Label{}`, I had to make the code a bit more verbose
```
➜  go run . # Before
[{Name:dagger.io/vcs.change.branch Value:} {Name:dagger.io/vcs.change.head_sha Value:} {Name:dagger.io/vcs.job.name Value:} {Name: Value:} {Name: Value:} {Name: Value:} {Name: Value:}]
➜  go run . # After
[{Name:dagger.io/vcs.change.branch Value:} {Name:dagger.io/vcs.change.head_sha Value:} {Name:dagger.io/vcs.job.name Value:}]
```